client/query/query.go

```diff
@@ -0,0 +1,159 @@
+package query
+
+import (
+	"fmt"
+
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+	"github.com/statechannels/go-nitro/client/engine/store"
+	"github.com/statechannels/go-nitro/payments"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// getStatusFromChannel returns the status of the channel
+func getStatusFromChannel(c *channel.Channel) ChannelStatus {
+	if c.FinalSignedByMe() {
+		if c.FinalCompleted() {
+			return Complete
+		}
+		return Closing
+	}
+	if !c.PostFundComplete() {
+		return Proposed
+	}
+	return Ready
+}
+
+// getPaymentChannelBalance generates a PaymentChannelBalance from the given participants and outcome
+func getPaymentChannelBalance(participants []types.Address, outcome outcome.Exit) PaymentChannelBalance {
+
+	numParticipants := len(participants)
+	// TODO: We assume single asset outcomes
+	sao := outcome[0]
+	asset := sao.Asset
+	payer := participants[0]
+	payee := participants[numParticipants-1]
+	paidSoFar := sao.Allocations[1].Amount
+	remaining := sao.Allocations[0].Amount
+	return PaymentChannelBalance{
+		AssetAddress:   asset,
+		Payer:          payer,
+		Payee:          payee,
+		PaidSoFar:      paidSoFar,
+		RemainingFunds: remaining,
+	}
+}
+
+// getLatestSupported returns the latest supported state of the channel
+// or the prefund state if no supported state exists
+func getLatestSupported(channel *channel.Channel) state.State {
+	if channel.HasSupportedState() {
+		supported, _ := channel.LatestSupportedState()
```

Comment:
Seems like the error is dropped here. Is that bad?

Reply:
I think I was just being a bit lazy since we'd never expect `LatestSupportedState` to  error if there is a supported state.

I've updated this in 4fee4032bd04201d3974adebc9c62ec66829a01c to bubble up the error.